src/components/flashcard/flashcard.tsx

```diff
@@ -86,4 +103,9 @@ export const Flashcard = ({
   function handleBackFaceChange(cardContent: CardContent) {
     onCardChange?.({ ...card, back: cardContent });
   }
+
+  function isCardTopInViewPort(cardDiv: HTMLDivElement) {
+    const { top } = cardDiv.getBoundingClientRect();
+    return top <= (window.innerHeight || document.documentElement.clientHeight);
```

Comment:
correctness wise, shouldn't we also check if `bottom > 0`?

but i guess it technically doesn't matter since new cards appear at the bottom

Reply:
also logic wise, this would mean (and what you have currently) that this returns true if the card **overlaps** instead of **in**

correct logic would be `top >= 0` and `bottom <= (window.innerHeight || document.documentElement.clientHeight)`

Reply:
I had it like this, but it was a little jarring because it scrolls to center, and often the cards are just a little off the screen, so it was too much movement. 

 I will try breaking the scroll into view into 2 cases: is too high and is too low. And it scrolls just enough to bring the entire card into view.

Reply:
Jk, that did not work at all. It will scroll just enough to get into view, so it is occluded by the header on the top and occluded by the add card button on the bottom if we do that. 

The original logic is it will only scroll if a card becomes active when completely outside of the viewport. I assumed this should only happen on `new card add` but I can add the reverse logic for the top for completeness.

Reply:
ah that makes perfect sense on that we only want to scroll if it's _completely_ out of the view (negation of any overlap)
left a comment on your most recent commit, once that's handled you can merge

sorry for the nits